### ISSUE TITLE
feature: Wrap commit body at 72 characters

### ISSUE THREAD

**Issue by jalil-salame** [jesseduffield/lazygit#2627]
**Is your feature request related to a problem? Please describe.**
I recently had some issues where I had to amend commits because their format was wrong; it was expected for the body to wrap at 72 characters.

**Describe the solution you'd like**
I would like the body of a multiline commit to warp at 72 characters.

**Describe alternatives you've considered**
- Set up a git hook to reject commit body lines longer than 72 characters
- Have a column at 72 characters in lazygit and manually wrap lines
- Fall back to the editor for multiline commits

**Additional context**
- Tim Pope on commit messages: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
- The 50/72 rule: https://stackoverflow.com/questions/2290016/git-commit-messages-50-72-formatting
- Kernel commit message format: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/submitting-patches.rst?id=bc7938deaca7f474918c41a0372a410049bd4e13#n664

**Comment by jesseduffield** [comment]
I like the idea of a ruler. I suspect it would be pretty straight-forward to implement

**Comment by mark2185** [comment]
Just pointing out that a [PR was in the works](https://github.com/jesseduffield/lazygit/pull/2013) for something like that.

**Comment by jalil-salame** [comment]
> Just pointing out that a [PR was in the works](https://github.com/jesseduffield/lazygit/pull/2013) for something like that.

This issue should probably be split into add a marker at X characters and wrap commit body at X characters, the sentiment appears to be that warping should be closed a won't fix or help wanted

**Comment by aubreypwd** [comment]
### Wrapping

It would also be nice to have an option we can set that will simply wrap the message where the box ends. It will still commit a long line, but the box itself will wrap. That way I can see my entire message.

![CleanShot 2023-08-23 at 10 08 23@2x](https://github.com/jesseduffield/lazygit/assets/1753298/63af8253-3ca8-4439-936c-8ae5602ffa9d)

⬆️  you can see I can't really see what I typed.

![CleanShot 2023-08-23 at 10 09 11@2x](https://github.com/jesseduffield/lazygit/assets/1753298/f424ff25-9beb-480e-a660-68aac287a877)

Ideally it would wrap the word, but the character would also be fine. Again, the commit message here in the reflog would be:

```
Update Rates to be affiliates with Affiliates

A Rate, before this commit, belonged to "Rates" but that isn't accurate, it should belong to Affiliates since they are the one's who have rates.
```

...without a break.

### Rulers

I also think Ruler settings would also be appropriate, we would need 2 (or more?):

One for subject, and one for the description.

**Comment by stefanhaller** [comment]
> It will still commit a long line, but the box itself will wrap.

I don't think this is desirable. It would encourage people to create commit messages with long lines, which is not a good idea. Commit messages should really be hard-wrapped so that they are readable on any git client.

> I also think Ruler settings would also be appropriate, we would need 2 (or more?):
> One for subject, and one for the description.

The subject already has a count for how many characters have been typed so far, so I don't think it needs a ruler. Only the body does.

**Comment by jwhitley** [comment]
In line with @stefanhaller's and @aubreypwd's comments, I think a rule at column 72 is probably ideal as a first step, but I'd take a column count too.  I've used git UIs that fool the user into thinking they hard-wrap, only later to discover commit messages which are unreadable in other tooling. So it's nice that lazygit is "WYSIWYG". OTOH, it's also annoying that there's zero formatting support, especially in projects that use hooks to enforce formatting rules. Once there's a column indicator, niceties like auto-word wrap at the end column could be added on top later.

**Comment by stefanhaller** [comment]
I disagree now. I know I argued for a ruler before, and against automatic wrapping, but I changed my mind. I started working on automatic wrapping-as-you-type (as shown [here](https://github.com/jesseduffield/lazygit/pull/3117#issuecomment-1817497882)), and I won't settle for less now. 😄 For me this is the perfect solution. We shouldn't have to do any manual interactions for wrapping our text to a certain width. You don't do that when typing a paragraph in Word, either.

I hope to have something working that can be tested soon.
